pyam/core.py

```diff
@@ -1838,13 +1849,16 @@ def subtract(self, a, b, name, axis="variable", fillna=None, append=False):
         :class:`IamDataFrame` or **None**
             Computed timeseries data or None if `append=True`.
         """
-        _value = _op_data(self, name, "subtract", axis=axis, fillna=fillna, a=a, b=b)
+        kwds = dict(axis=axis, fillna=fillna, ignore_units=ignore_units)
+        _value = _op_data(self, name, "subtract", **kwds, a=a, b=b)
         if append:
             self.append(_value, inplace=True)
         else:
             return IamDataFrame(_value, meta=self.meta)
 
-    def multiply(self, a, b, name, axis="variable", fillna=None, append=False):
+    def multiply(
+        self, a, b, name, axis="variable", fillna=None, ignore_units=False, append=False
+    ):
         """Compute the product of timeseries data between `a` and `b` along an `axis`
 
         This function computes `a * b`. If `a` or `b` are lists, the method applies
```

Comment:
Should we add a note in the description here and for the divide-function that units might be reformatted by pint? For addition and subtraction this should not be necessary.
Also I was wondering which unit pint returns when multiplying e.g. `PJ` with `1 / PJ`. - Does it recognise it as `dimensionless` or will it return `PJ / PJ`? This might become important when using more complicated functions in `apply` - so we can keep it in mind to test in a following PR.

Reply:
Added a docstring to all four binary-ops functions and `apply()` - after all, as soon as not all units are identical, all functions will revert to abbreviated-units style of pint.

About more elaborate operations corner cases like you describe, my experience is that pint is quite good at dealing with those...

Reply:
Addition and subtraction should lead to an error, if it's not done with identical units.

I think you are right about pint that it recognizes `PJ / PJ`. What it doesn't recognize are such things like `TWh / PJ` - but I would leave that to the user to convert it into `dimensionsless`.

Reply:
> Addition and subtraction should lead to an error, if it's not done with identical units.

Why? If you add PJ and EJ or Mt CO2 and Gt CO2, the units are compatible so why should it error? 

> What it doesn't recognize are such things like `TWh / PJ`

Pint is generally fine with this. E.g. if I had W * s / J then you could convert to dimensionless and it should all work without an issue. Whether pint knows about TWh or not is a different question (is TWh defined in pint or iam_units?)

Reply:
> Why? If you add PJ and EJ or Mt CO2 and Gt CO2, the units are compatible so why should it error?

I haven't thought about that :D I was thinking about different units like EJ and Mt CO2 and pint should raise and error then, hopefully ;-)

> Pint is generally fine with this. E.g. if I had W * s / J then you could convert to dimensionless and it should all work without an issue. Whether pint knows about TWh or not is a different question (is TWh defined in pint or iam_units?)

I just tried pint with meters and inch and it returned meter/inch, which I had to convert (using pint) into dimensionless later. But I don't see an issue with that - users shout convert the units either before or after by themselves.